f/equalsAsync.js

```diff
@@ -0,0 +1,30 @@
+/**
+ *
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+async function equals(it1, it2) {
+  it1 = it1[Symbol.iterator]();
+  it2 = it2[Symbol.iterator]();
+  for await (;;) {
```

Comment:
Sadly, this is not how `for await` works 😅 The only implicit await construct is `for await(let x of asyncIterator) { ... }`.

In this case, you’d probably use a normal `while(true)` and say `let i1 = await it1.next(), ...`.